src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -924,6 +924,120 @@ private with sharing class RD2_OpportunityService_TEST {
         assertPaymentIsMatchingOpp(oppById);
     }
 
+    /***
+    * @description Verifies if Account is not getting removed in opportunity
+    * when recurring donation record with only contact is updated.
+    */
+    @isTest
+    private static void shouldNotRemoveAccountInOppWhenRDWithNoAccountIsModified() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        setUpConfiguration();
+
+
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        //create opp from rd with contact only
+        Opportunity opp = getOpportunityBuilderWithNoAccount(rd)
+            .withCloseDate(CLOSE_DATE)
+            .withOpenStage()
+            .build();
+        insert opp;
+
+
+        //Verify if accountid is populated in opp when only contact is provided in RD
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+        System.assertEquals(1, opps.size(), 'One Installment Opp should be created on RD insert: ' + opps);
+        System.assert(opps[0].AccountId != null);
+
+        //update rd amount
+        Test.startTest();
+        rd.npe03__Amount__c = RD_NEW_AMOUNT;
+        update rd;
+        Test.stopTest();
+
+        //Verify if accountid is not nulled out when rd record is modified
+        List<Opportunity> oppUpdated = oppGateway.getRecords(rd);
```

Comment:
No need for the new var, the `opps` var can be reused. Based on the comments above, the test with updated ApexDoc, the test name, spacing and assert statements would be as:
```
    /***
    * @description Verifies the Account on Opportunity is not nulled
    * when Recurring Donation having Contact specified only is updated
    */
    @isTest
    private static void shouldNotRemoveAccountOnOppWhenRDWithoutAccountIsModified() {
        RD2_EnablementService_TEST.setRecurringDonations2Enabled();

        setUpConfiguration();

        npe03__Recurring_Donation__c rd = createRecurringDonation();

        //create a related Opp where Contact is specified only
        Opportunity opp = new TEST_OpportunityBuilder()
            .withContact(rd.npe03__Contact__c)
            .withRecurringDonation(rd.Id)
            .withAmount(rd.npe03__Amount__c)
            .withCloseDate(CLOSE_DATE)
            .withOpenStage()
            .build();
        insert opp;

        List<Opportunity> opps = oppGateway.getRecords(rd);
        System.assertNotEquals(null, opps[0].AccountId, 
            'Opp Account should be populated on the Opp insert');

        Test.startTest();
        rd.npe03__Amount__c = RD_NEW_AMOUNT;
        update rd;
        Test.stopTest();

        rd = rdGateway.getRecord(rd.Id);
        opps = oppGateway.getRecords(rd);

        System.assertEquals(1, opps.size(), 'One Opp should exist: ' + opps);
        System.assertEquals(RD_NEW_AMOUNT, opps[0].Amount, 'Opp Amount should be updated');
        System.assertEquals(getContact().AccountId, opps[0].AccountId, 
            'Opp Account should be the same as the Contact household Account');
    }
```